src/NonPhpFile/NetteDINeonMethodCallRenamer.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Nette\NonPhpFile;
+
+use Nette\Utils\Strings;
+use Rector\Core\Contract\Processor\FileProcessorInterface;
+use Rector\Core\ValueObject\Application\File;
+use Rector\Renaming\Collector\MethodCallRenameCollector;
+
+/**
+ * @see \Rector\Nette\Tests\NonPhpFile\NetteDINeonMethodCallRenamer\NetteDINeonMethodCallRenamerTest
+ */
+final class NetteDINeonMethodCallRenamer implements FileProcessorInterface
+{
+    /**
+     * @var MethodCallRenameCollector
+     */
+    private $methodCallRenameCollector;
+
+    public function __construct(MethodCallRenameCollector $methodCallRenameCollector)
+    {
+        $this->methodCallRenameCollector = $methodCallRenameCollector;
+    }
+
+    /**
+     * @param File[] $files
+     */
+    public function process(array $files): void
+    {
+        foreach ($files as $file) {
+            $this->processFile($file);
+        }
+    }
+
+    public function supports(File $file): bool
+    {
+        $fileInfo = $file->getSmartFileInfo();
+        return $fileInfo->hasSuffixes($this->getSupportedFileExtensions());
+    }
+
+    public function getSupportedFileExtensions(): array
+    {
+        return ['neon'];
+    }
+
+    private function processFile(File $file): void
+    {
+        $content = $file->getFileContent();
+        foreach ($this->methodCallRenameCollector->getMethodCallRenames() as $methodCallRename) {
+            $oldObjectType = $methodCallRename->getOldObjectType();
+            $objectClassName = $oldObjectType->getClassName();
+            $className = str_replace('\\', '\\\\', $objectClassName);
+
+            $oldMethodName = $methodCallRename->getOldMethod();
+            $newMethodName = $methodCallRename->getNewMethod();
+
+            $pattern = '/\n(.*?)(class|factory): ' . $className . '(\n|\((.*?)\)\n)\1setup:(.*?)- ' . $oldMethodName . '\(/s';
+            while ($matches = Strings::match($content, $pattern)) {
+                $replacedMatch = str_replace($oldMethodName . '(', $newMethodName . '(', $matches[0]);
+                $content = str_replace($matches[0], $replacedMatch, $content);
+            }
+        }
+        $file->changeFileContent($content);
```

Comment:
As we spoke before, here should be only iteration of classes the implement `RectorInterface`, not the change itself.

E.g. `class ConfigMethodRenamerRector implements RectorInterface { ... }`

Reply:
But now it is invoked by `RenameMethodRector` same as the old NonPhpFileProcessor is invoked by `RenameClassRector`.
If there will be some `ConfigMethodRenameRector` we will configure method renaming in two places? Or am I missing something (again)?

Reply:
The `RenameClassRector` is not good example. Both should be refactored towards Rector rule design.

Reply:
OK here is my suggestion. This works pretty well with no additional configuration needed. Let's merge it, keep it live for a while (maybe prepare some symfony / laravel / cake alternatives), document them, collect feedback and then discuss how to refactor them all.

Reply:
That would be a quick fix here. But it would introduce more ambiguous design, that leads to this in the first place. We should go the other way instead: make file processors the dispatchers and rules the subscribers.

I can look into it next week or two.

Reply:
Yes please. Look into it :) You have some idea, I can test it and verify if you want.

Reply:
Maybe @sabbelasichon would like to join us in this process